test/functional/test_framework/test_node.py

```diff
@@ -85,24 +85,33 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, timeout_factor, bitc
         self.coverage_dir = coverage_dir
         self.cwd = cwd
         self.descriptors = descriptors
+        self.init_rpc_url = rpc_url(self.datadir, self.index, self.chain, self.rpchost)
         if extra_conf is not None:
             append_config(datadir, extra_conf)
         # Most callers will just need to add extra args to the standard list below.
         # For those callers that need more flexibility, they can just set the args property directly.
         # Note that common args are set in the config file (see initialize_datadir)
         self.extra_args = extra_args
         self.version = version
+
+        # Calculate RPC address to be passed into the command line
+        rpc_addr = self.init_rpc_url.split("http://")[-1].split('@')[-1]
+
         # Configuration for logging is set as command-line args rather than in the bitcoin.conf file.
         # This means that starting a bitcoind using the temp dir to debug a failed test won't
         # spam debug.log.
         self.args = [
             self.binary,
-            "-datadir=" + self.datadir,
-            "-logtimemicros",
-            "-debug",
-            "-debugexclude=libevent",
-            "-debugexclude=leveldb",
-            "-uacomment=testnode%d" % i,
+            # TODO: change the following to regtest once we have one
+            "--net=mainnet",
+            "--rpc-addr={}".format(rpc_addr),
+            #"-X",
+            #"-datadir=" + self.datadir,
+            #"-logtimemicros",
+            #"-debug",
+            #"-debugexclude=libevent",
+            #"-debugexclude=leveldb",
+            #"-uacomment=testnode%d" % i,
         ]
         if use_valgrind:
```

Comment:
I think this can be removed. I don't think we need to have valgrind code in the testing script, not at least right now.